Code/Framework/AzFramework/AzFramework/TargetManagement/TargetManagementComponent.cpp

```diff
@@ -444,31 +350,144 @@ namespace AzFramework
-
-        // Until an authentication connection can be established between peers only supporting
-        // local connections (i.e. binding to localhost).
-        searchParams.m_listenAddress = "127.0.0.1";
-        searchParams.m_serverAddress = "127.0.0.1";
-
-        EBUS_EVENT_ID_RESULT(m_networkImpl->m_gridSearch,m_networkImpl->m_gridMate,GridMate::LANSessionServiceBus,StartGridSearch,searchParams);
-        if (m_networkImpl->m_gridSearch)
+        // If we're a client, set the host to our desired target
+        if (connection->GetConnectionRole() == AzNetworking::ConnectionRole::Connector)
         {
-            m_reconnectionTime = AZStd::chrono::system_clock::now() + AZStd::chrono::milliseconds(m_settings->m_reconnectionIntervalMS);
+            if (GetTargetInfo(packet.GetPersistentId()).GetPersistentId() == 0)
+            {
+                TargetContainer::pair_iter_bool ret = m_availableTargets.insert_key(packet.GetPersistentId());
+
+                TargetInfo& ti = ret.first->second;
+                ti.m_lastSeen = time(nullptr);
+                ti.m_displayName = "Host";
+                ti.m_persistentId = packet.GetPersistentId();
+                ti.m_networkId = static_cast<uint32_t>(connection->GetConnectionId());
+                ti.m_flags |= TF_ONLINE | TF_DEBUGGABLE | TF_RUNNABLE;
+
+                // last target can be loaded in with a persistent id but no network, correct here
+                if (GetDesiredTarget().GetPersistentId() == packet.GetPersistentId())
+                {
+                    m_settings->m_lastTarget.m_networkId = ti.m_networkId;
+                }
+
+                EBUS_QUEUE_EVENT(TargetManagerClient::Bus, TargetJoinedNetwork, ti);
+            }
+
+            if (GetDesiredTarget().GetPersistentId() != packet.GetPersistentId())
+            {
+                SetDesiredTarget(packet.GetPersistentId());
+            }
+        }
+
+        const uint32_t totalBufferSize = packet.GetSize();
+
+        // Messages can be larger than the size of a packet so reserve a buffer for the total message size
+        if (m_tmpInboundBufferPos == 0)
+        {
+            m_tmpInboundBuffer.reserve(totalBufferSize);
+        }
+
+        // Read as much data as the packet can include and append it to the buffer
+        const uint32_t readSize = AZStd::min(totalBufferSize - m_tmpInboundBufferPos, Neighborhood::NeighborBufferSize);
+        memcpy(m_tmpInboundBuffer.begin() + m_tmpInboundBufferPos, packet.GetMessageBuffer().GetBuffer(), readSize);
+        m_tmpInboundBufferPos += readSize;
+        if (m_tmpInboundBufferPos == totalBufferSize)
+        {
+            // Deserialize the complete buffer
+            AZ::IO::MemoryStream msgBuffer(m_tmpInboundBuffer.data(), totalBufferSize, totalBufferSize);
+            TmMsg* msg = nullptr;
+            AZ::ObjectStream::ClassReadyCB readyCB(AZStd::bind(&TargetManagementComponent::OnMsgParsed, this, &msg, AZStd::placeholders::_1,
```

Comment:
A lambda should be preferred over using [std::bind](https://abseil.io/tips/108).

Also in this case the TmMsg can be initialized with AZ::Utils::LoadObjectFromStreamInPlace](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Serialization/Utils.h#L68-L72)

Reply:
Actually, this code was only moved.
Nevermind.

Reply:
Thanks, I'll take a look.